tests/drop_test.py

```diff
@@ -0,0 +1,133 @@
+                                        tags=['pushable', 'pickable'])
+        expected_dummy_object4 = Object(id=expected_scene.generate_id(), name="Stone", position=Vector(5, 6),
+                                        tags=['pushable', 'pickable'])
+        expected_scene.add_entity(expected_dummy_object)
+        expected_scene.add_entity(expected_dummy_drop)
+        expected_scene.add_entity(expected_dummy_object2)
+        expected_scene.add_entity(expected_dummy_object3)
+        expected_scene.add_entity(expected_dummy_object4)
+
+        random.seed(1222)
+        scene = Scene(players=[Player(player_id="dummy", player_resources={"Wood": 0, "Stone": 0}, drop_area=Vector(5, 5))])
+        dummy_object = Object(id=1, name="dummy", owner="dummy", position=Vector(5, 5), inventory={"Wood": 2, "Stone": 1})
+        dummy_drop = Drop(target_position=Vector(5, 6), object_id=1)
+        scene.add_entity(dummy_object)
+        scene.add_entity(dummy_drop)
+
+        dummy_drop.apply(scene)
+        for action in scene.pending_actions:
+            Action.deserialize(action).apply(scene)
+        scene.pending_actions.clear()
+
+        assert scene == expected_scene
+
+    def test_success_drop_to_drop_area_apply(self):
+        expected_scene = Scene(players=[Player(player_id="test", player_resources={"Wood": 2, "Stone": 1}, drop_area=Vector(5, 6))])
+        expected_dummy_object = Object(id=1, name="dummy", owner="test", position=Vector(5, 5))
+        expected_dummy_drop = Drop(target_position=Vector(5, 6), object_id=1)
+        expected_update_points = UpdateResources(user_name="test", points_updated = {"Wood": 2, "Stone": 1})
+        expected_scene.add_entity(expected_dummy_object)
+        expected_scene.add_entity(expected_dummy_drop)
+        expected_scene.pending_actions.append(expected_update_points.serialize())
+
+        scene = Scene(players=[Player(player_id="test", player_resources={"Wood": 0, "Stone": 0}, drop_area=Vector(5, 6))])
+        dummy_object = Object(id=1, name="dummy", owner="test", position=Vector(5, 5), inventory={"Wood": 2, "Stone": 1})
+        dummy_drop = Drop(target_position=Vector(5, 6), object_id=1)
+        scene.add_entity(dummy_object)
+        scene.add_entity(dummy_drop)
+
+        dummy_drop.apply(scene)
+
+        assert scene == expected_scene
+class TestDropRequirements:
+
+    def test_success_requirements(self):
+        scene = Scene(players=[Player(player_id="dummy", player_resources={"Wood": 0, "Stone": 0}, drop_area=Vector(5, 5))])
+        dummy_object = Object(id=1, name="dummy", owner="dummy", position=Vector(5, 5), inventory={"Wood": 1})
+        dummy_drop = Drop(target_position=Vector(5, 6), object_id=1)
+        scene.add_entity(Object(id=3, name="Grass", position=Vector(5, 6), tags=['walkable']))
+        scene.add_entity(dummy_object)
+        assert dummy_drop.satisfies_requirements(scene) is True
+
+    def test_fail_requirements_distance(self):
```

Comment:
This test actually checks what happens when there's no `Tile` to `Drop` on
`test_no_tile_to_drop_on_negative/fails`